src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient_.cs

```diff
@@ -51,62 +187,47 @@ internal partial class ImageStoreClient
+            }
+            else
+            {
+                pathInImageStore.ThrowIfNull(nameof(fileContentsToUpload));
+                pathInImageStore.ThrowIfNull(nameof(pathInImageStore));
+                serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
+                var url = "ImageStore/{pathInImageStore}";
+                url = url.Replace("{pathInImageStore}", Uri.EscapeDataString(pathInImageStore.ToString()));
+                requestId = requestId ?? Guid.NewGuid().ToString();
+                var queryParams = new List<string>();
 
-            // Append to queryParams if not null.
-            serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
-            queryParams.Add("api-version=6.0");
-            url += "?" + string.Join("&", queryParams);
+                // Append to queryParams if not null.
+                serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
+                queryParams.Add("api-version=6.0");
+                url += "?" + string.Join("&", queryParams);
 
-            HttpRequestMessage RequestFunc()
-            {
-                var request = new HttpRequestMessage()
+                HttpRequestMessage RequestFunc()
                 {
-                    Method = HttpMethod.Put,
-                    Content = new ByteArrayContent(fileContentsToUpload)
-                };
-
-                request.Content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/octet-stream");
-                return request;
-            }
+                    var request = new HttpRequestMessage()
+                    {
+                        Method = HttpMethod.Put,
+                        Content = new ByteArrayContent(fileContentsToUpload)
+                    };
 
-            return this.httpClient.SendAsync(RequestFunc, url, requestId, cancellationToken);
-        }
+                    request.Content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/octet-stream");
+                    return request;
+                }
 
-        /// <inheritdoc />
-        public Task UploadFileChunkAsync(
-            byte[] fileChunkToUpload,
-            string pathInImageStore,
-            Guid? sessionId,
-            long startBytePosition,
-            long endBytePosition,
-            long length,
-            long? serverTimeout = 60,
-            CancellationToken cancellationToken = default(CancellationToken))
-        {
-            return this.UploadFileChunkAsync(
-                fileChunkToUpload,
-                pathInImageStore,
-                sessionId,
-                endBytePosition,
-                endBytePosition,
-                length,
-                null,
-                serverTimeout,
-                cancellationToken);
+                await this.httpClient.SendAsync(RequestFunc, url, requestId, cancellationToken);
```

Comment:
>await [](start = 16, length = 5)

This should just be return, you dont need to await it.
[https://stackoverflow.com/questions/21033150/any-difference-between-await-task-run-return-and-return-task-run](https://stackoverflow.com/questions/21033150/any-difference-between-await-task-run-return-and-return-task-run)

Reply:
You can't return a task in an async method, and the method needs to be async since it awaits getting the connection string as I mentioned previously 

`error CS1997: Since 'ImageStoreClient.UploadFileAsync(byte[], string, string, long?, CancellationToken)' is an async method that returns 'Task', a return keyword must not be followed by an object expression. Did you intend to return 'Task<T>'?`

Reply:
It's fine in this case, there were other methods in this PR which could have just returned task, please use this as a general guidance wherever applicable.